src/main/scala/io/iohk/ethereum/domain/branch/BlockchainBranch.scala

```diff
@@ -0,0 +1,16 @@
+package io.iohk.ethereum.domain.branch
+
+import akka.util.ByteString
+
+import io.iohk.ethereum.domain.Block
+
+// TODO choose a name : ChainInstance, BlockchainBranch, Branch, Blockchain ?
+trait BlockchainBranch {
```

Comment:
is it gonna be used only for the main chain? or we could be interested in alternative branches?

Reply:
The intent is to be able to use it for other chains as well. So the interface itself does not make any assumption that we are indeed using the best chain.

Reply:
then maybe just Branch as a name will suit better, that'd be my choice

Reply:
OK. Let's go for it for now. This part is still heavily changing right now so we will see how this pans out.